src/devtools/domains/history.js

```diff
@@ -17,9 +17,19 @@ class History {
 
     while (cursor) {
       list.push(cursor)
-      cursor = cursor.children.find(child => {
+
+      // cache current action
+      parent = cursor
+
+      // set the cursor to the action's next child
+      cursor = parent.children.find(child => {
         return child.id === cursor.next
       })
+
+      // if no next child was found but children exist, pick one
+      if (!cursor && parent.children.length) {
+        cursor = parent.children[0]
+      }
```

Comment:
Looks like we push to the end of the children list https://github.com/vigetlabs/microcosm/blob/master/src/action.js#L232, should we pick the last child?